core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -63,7 +63,6 @@ object RequestChannel extends Logging {
     // o.a.k.common.requests.AbstractRequest.getRequest()
     private val keyToNameAndDeserializerMap: Map[Short, (ByteBuffer) => RequestOrResponse]=
       Map(ApiKeys.FETCH.id -> FetchRequest.readFrom,
-        ApiKeys.METADATA.id -> TopicMetadataRequest.readFrom,
```

Comment:
I think `TopicMetadataRequest.readFrom` is not used anymore outside of tests after this. Remove it?